packages/rmb_direct_client/lib/client.ts

```diff
@@ -167,9 +167,27 @@ class Client {
     }
   }
 
-  disconnect() {
+  private async waitForResponses(): Promise<void> {
+    const start = new Date().getTime();
+    while (new Date().getTime() < start + 2 * 60 * 1000) {
+      if (this.responses.size === 0) return;
+      console.debug("Waiting for the rmb responses to be received before closing the connection");
+      for (const id of this.responses.keys()) {
+        const envelope = this.responses.get(id);
+        if (envelope?.request) {
+          console.debug(`- Response for ${envelope?.request.command} from twin ${envelope?.destination}`);
+        }
+      }
```

Comment:
Make the code more readable by breaking down the logic into smaller functions. For example, you could have a separate function for processing responses and another for logging.


```suggestion
  const logResponses = () => {
    for (const [id, envelope] of this.responses) {
      if (envelope?.request) {
        console.debug(`- Response for ${envelope.request.command} from twin ${envelope.destination}`);
      }
    }
  };
  
  const logMessageBeforeResponse = () => {
      console.debug("Waiting for the rmb responses to be received before closing the connection");
  }

  while (new Date().getTime() < maxTimeout) {
    if (this.responses.size === 0) return;
    logMessageBeforeResponse();
    logResponses();
    await new Promise((resolve) => setTimeout(resolve, delay));
  }
```